pywbem_mock/_cimrepository.py

```diff
@@ -2485,236 +2521,295 @@ def _pull_operations_disabled(self):
 
-        return self._open_response(objects, namespace,
-                                   'PullInstancesWithPath', **params)
+        return self._open_response(instances, namespace,
+                                   'PullInstancesWithPath',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   OpenQueryInstances server request
     #
     ####################################################################
 
-    def OpenQueryInstances(self, **params):
+    def OpenQueryInstances(self, FilterQueryLanguage, FilterQuery,
+                           namespace=None, ReturnQueryResultClass=None,
+                           OperationTimeout=None, ContinueOnError=None,
+                           MaxObjectCount=None):
         # pylint: disable=invalid-name
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenQueryInstances`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
         self.validate_namespace(namespace)
-        self._validate_open_params(**params)
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout)
 
         # pylint: disable=assignment-from-no-return
         # Issue #2064 TODO/ks implement execquery
-        result = self.ExecQuery(**params)
-
-        objects = [] if result is None else [x[2] for x in result[0][2]]
+        # Handle ReturnQueryResultClass maybe or make it exception
+        instances = self.ExecQuery(FilterQueryLanguage, FilterQuery,
+                                   namespace=namespace)
 
-        return self._open_response(objects, namespace,
-                                   'PullInstancesWithPath', **params)
+        return self._open_response(instances, namespace,
+                                   'PullInstancesWithPath',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   PullInstancesWithPath server request
     #
     ####################################################################
 
-    def PullInstancesWithPath(self, **params):
+    def PullInstancesWithPath(self, EnumerationContext, MaxObjectCount):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenPullInstancesWithPath`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
```

Comment:
See previous comment about `namespace = self.namespace`. Only this time, the namespace should be available in EnumerationContext.